### DISCUSSION TITLE
CouchbaseContainer - Use the official Docker image to have the data samples in the cluster

### DISCUSSION THREAD

**Discussion by Cyrille-Dakhlia** [testcontainers/testcontainers-java#4657] [Q&A]
Hi there! 👋

I have discovered `TestContainers` a few days ago as I am currently working with Couchbase and wanting to test my development, and I have found that there is a Couchbase module in here which is great news to me!

I would like to run a CouchbaseContainer from the **official Docker image to have all the data samples** in it, but I am not quite sure how to do that. It looks like the following call does not use the official Couchbase Docker image by default:
`public static CouchbaseContainer couchbaseContainer = new CouchbaseContainer("couchbase/server");`

Am I missing something or is it intended that the cluster in the container is empty?

**Another note:**
I tried to retrieve a bucket that **doesn't exist** in the cluster, and to list its scopes, and I got the following error:
`The cluster does not support collections and scopes.
com.couchbase.client.core.error.FeatureNotAvailableException: The cluster does not support collections and scopes.`
It seems to me that the error message is not correct since, when adding a bucket to the cluster, I can effectively retrieve its scopes, which means that the cluster **does** support that feature.
I don't know if it is intended as well, but it was misleading to me and I thought that probably `TestContainers` was actually not handling Scopes and Collections features as this feature is very new (July 2021 - version 7.0)

Thanks in advance for any answers!

Kind regards,
Cyrille Dakhlia

**Comment by daschl** [comment]
@Cyrille-Dakhlia are you referring to being able to load the sample buckets like travel-sample, beer-sample etc? If so I had a change somewhere locally - if there is interest I can dust it up and bring it up as a PR.

The CouchbaseContainer supports 7.0 but you need to create scopes and collections manually through the SDK management APIs at this point. I'm planning to add support for collections and scopes soon so that it's going to be even easier going forward.

For the error message, can you show me the setup and SDK code you used so I can get a closer look? Also I'd like to know which SDK version you used exactly for a potential repro.

**Reply by Cyrille-Dakhlia-Zeenea** [reply]
Hi @daschl , thank you for your answer!

Indeed, I was referring to those sample buckets ("travel-sample", "beer-sample", etc.). But I didn't use them in the end because it was a lot to load, so I chose the option to create my own simple samples 👍 ^^ So there is no need (at least for me) anymore, thank you for your proposition.

I did create manually collections and scopes and it enabled me to move forward, but support for that would always be good I think to make it easier as you said! 👍

About the error message, I am using `TestContainers version 1.16.2` and I'm not quite sure what I had at that time... but I think that I simply created a new CouchebaseContainer with a new BucketDefinition, started it and tried to access scopes from that bucket, leading to that error since I did not created any scopes at that moment. And showing me that the cluster was not "supporting" collections and scopes made me think the feature was not available, but in the end, creating scopes and collections enabled me to access them and not have that error message anymore.

**Comment by Cyrille-Dakhlia-Zeenea** [comment]
I am facing another issue,
When creating a new CouchbaseContainer and starting it to generate data in it for my tests, I have an odd `ContainerLaunchException` raised because of a `SocketTimeoutException` that appears *randomly* it seems... I have tried to tweak my CouchbaseContainer settings using the following methods:
```
this.start();
this.withStartupTimeout(Duration.ofSeconds(120));
this.waitUntilContainerStarted();`
but I still randomly have the exception.

I still use version 1.16.2 of TestContainers.

I have let this issue aside but I will have to get back to it soon.

**Comment by daschl** [comment]
@Cyrille-Dakhlia-Zeenea would it be possible to enable debug logging and share those logs? The reason is I added timings to the individual phases and maybe that gives us a better clue where the time is spent.
